script.js

```diff
@@ -0,0 +1,203 @@
+    ingredients: [
+      { NAME: "potatoes", AMOUNT: "2 cups" },
+      { NAME: "peas", AMOUNT: "1 cups" },
+      { NAME: "spices", AMOUNT: "To taste" },
+      { NAME: "flour", AMOUNT: "2 cups" },
+      { NAME: "water", AMOUNT: "1/2 cup" },
+    ],
+    instructions:
+      " Prepare the filling with boiled potatoes, peas, and spices and make a dough using flour and water. Roll out the dough and fill it with the prepared mixture. Shape the samosas and deep-fry until golden brown.",
+  },
+  {
+    id: 6,
+    title: "Gulab Jamun",
+    picture_url:
+      "https://www.google.com/imgres?imgurl=https%3A%2F%2Fc2.staticflickr.com%2F6%2F5559%2F30562381266_8994c463dd_z.jpg&tbnid=1byBguk9ArC-sM&vet=12ahUKEwiF_POvjNaDAxUqDBAIHXA9CQoQMygcegUIARDJAQ..i&imgrefurl=https%3A%2F%2Fwww.chitrasfoodbook.com%2F2016%2F10%2Fgulab-jamun-recipe-using-instant-mix.html&docid=kgSj79Z8hJliDM&w=499&h=640&q=image%20of%20gulab%20jamun&client=firefox-b-d&ved=2ahUKEwiF_POvjNaDAxUqDBAIHXA9CQoQMygcegUIARDJAQ",
+    ingredients: [
+      { NAME: "milk powder", AMOUNT: "1 cup" },
+      { NAME: "ghee", AMOUNT: "2 tablespoons" },
+      { NAME: "milk", AMOUNT: "1/4 cup" },
+      { NAME: "sugar", AMOUNT: "1 cup" },
+      { NAME: "cardamom", AMOUNT: "1/2 teaspoon" },
+    ],
+    instructions:
+      "Mix milk powder, ghee, and milk to form a dough. Shape the dough into small balls. Deep-fry the balls until golden brown. Prepare sugar syrup with sugar and cardamom. Soak the fried balls in the sugar syrup.",
+  },
+];
+
+// Create new recipe
+function createRecipeElement(recipe) {
+  const container = document.getElementById("recipe-container");
+  const recipeCard = document.createElement("div");
+  recipeCard.className = "food-card";
+  const titleElement = document.createElement("h2");
+  titleElement.textContent = recipe.title;
+
+  const ingredientsElement = document.createElement("ul");
+  recipe.ingredients.forEach((ingredient) => {
+    const li = document.createElement("li");
+    li.textContent = `${ingredient.NAME}: ${
+      ingredient.AMOUNT || "Not specified"
+    }`;
+    ingredientsElement.appendChild(li);
+  });
+
+  //creating instruction for receipe
+  const instructionsElement = document.createElement("p");
+  instructionsElement.textContent = recipe.instructions;
+
+  recipeCard.appendChild(titleElement);
+  recipeCard.appendChild(ingredientsElement);
+  recipeCard.appendChild(instructionsElement);
+  container.appendChild(recipeCard);
+}
+function addIngredient(recipes, newIngredient) {
+  recipe.ingredients.push({ NAME: newIngredient, AMOUNT: undefined });
```

Comment:
I will suggest to give a value to amount or just remove it if you anyway don't use it :)